dali/base/dasds.cpp

```diff
@@ -3675,6 +3790,78 @@ class CLockInfo : public CInterface, implements IInterface
         return out;
     }
 
+    bool getFilteredLocks(CLockFilters &filters, MemoryBuffer &mb)
+    {
+        bool isFile;
+        StringBuffer file;
+        if (!filters.checkAndReadPath(xpath, isFile, file))
```

Comment:
The xpath filter should be like other filters, optional and configurable, so could say, e.g.:
xpath=/Files/Scope[@name="thor400"]|endpoint=..

You can construct that filter based on the esp request, i.e. whether there's a xpath=/Files filter or not based getFileLockOnly()

I'm not sure you need your DALTFNonFileLock and DALTFFileLock types in that case, but other 'lock types' might be useful, e.g. write locks vs read locks.